### ISSUE TITLE
Project configuration

### ISSUE THREAD

**Issue by ealeksandrov** [theappbusiness/ConfigGenerator#21]
> Finally, you need to create a build target for each of your enviroments. This can be done be selecting File -> New -> Target and selecting 'External Build System' 
> 
> The best way to support multiple environments is to define a separate scheme for each one. Then add the relevant target as an external build step for each scheme ensuring that 'Parallelize Build' is disabled.

Why is this so complex? I found adding `build pre-action` in scheme settings working just fine:
<img width="895" alt="screen shot 2017-07-14 at 15 11 17" src="https://user-images.githubusercontent.com/3132438/28211763-02739344-68a7-11e7-80e7-2d07d7223098.png">

Am I missing something?

**Comment by samdods** [comment]
@ealeksandrov have you verified whether this works when building from the command line?

**Comment by ealeksandrov** [comment]
@samdods yep, it generates different configs when using `xcodebuild` with different `-scheme` option.
Important part here is to mark scheme as `shared` to save script in source control and reuse inside dev team and CI.

**Comment by samdods** [comment]
Thanks, I'll update the readme and the blogpost

**Comment by ealeksandrov** [comment]
Cool! You can also mention `configen` availability on homebrew - #22.

**Comment by KaneCheshire** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Issue #21: Project configuration](https://trello.com/c/ktVFEBhG/18-issue-21-project-configuration)

**Comment by samdods** [comment]
@ealeksandrov, when using your configuration, I don't seem to get build errors when there's a problem.

**Comment by ealeksandrov** [comment]
@samdods output (stdout/stderr) from scheme scripts are not redirected into Xcode.

But function output can be checked and handled in same script like:
```sh
configen -p ${SRCROOT}/Configs/prod-config.plist -h ${SRCROOT}/Configs/AppEnvironment.map -n AppConfig -o ${SRCROOT}/Configs

if [ $? -ne 0 ]
then rm ${SRCROOT}/Configs/AppConfig.swift
fi
```
I can confirm it's working for me. If any error arises - `AppConfig.swift` will be removed to trigger compile-time error.
Another good idea is to append/replace this file with error string, Xcode will stop on it and present to user. But it will require little more bash-magic. And maybe this should go into `configen` itself since it already generates this file?
Not sure about your direction and vision here. Let me know how can I help!
